### PR TITLE
AMD-Pensando:Enable ionic driver compilation for x86

### PR DESCRIPTION
**Pull request by shanshri**
This patchset enables ionic driver compilation for AMD-Pensando ELBA SOC.
Ionic driver is an ethernet driver used for mgmt and etherenet interface if ELBA SOC.

The changes are verified on Pensando DSC2-200 card.
Tested by pinging from card to host and vice versa.
```
root@Lipari9:/home/admin# lspci -d 1dd8:
01:00.0 Memory controller: Pensando Systems Device 0009
02:00.0 Memory controller: Pensando Systems Device 000a
06:00.0 PCI bridge: Pensando Systems Device 0002
07:00.0 PCI bridge: Pensando Systems DSC Virtual Downstream Port
08:00.0 Ethernet controller: Pensando Systems DSC Management Controller
08:00.1 Serial controller: Pensando Systems Device 100a
root@Lipari9:/home/admin# dmesg | grep -i ionic
[  408.227143] ionic 0000:08:00.0: enabling device (0000 -> 0002)
[  408.227259] ionic 0000:08:00.0: 15.752 Gb/s available PCIe bandwidth, limited by 8.0 GT/s PCIe x2 link at 0000:00:03.1 (capable of 31.504 Gb/s with 8.0 GT/s PCIe x4 link)
[  413.264160] ionic 0000:08:00.0 eth1: Link up - 1 Gbps
root@Lipari9:/home/admin# ifconfig eth1
eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.8.2  netmask 255.255.255.0  broadcast 169.254.8.255
        inet6 fe80::690:81ff:fe00:1226  prefixlen 64  scopeid 0x20<link>
        ether 04:90:81:00:12:26  txqueuelen 1000  (Ethernet)
        RX packets 9  bytes 770 (770.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 14  bytes 1332 (1.3 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

root@Lipari9:/home/admin# ping 169.254.8.1
PING 169.254.8.1 (169.254.8.1) 56(84) bytes of data.
64 bytes from 169.254.8.1: icmp_seq=1 ttl=64 time=0.439 ms
64 bytes from 169.254.8.1: icmp_seq=2 ttl=64 time=0.185 ms
--------------------------------------------------------------
int_mnic0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.8.1  netmask 255.255.255.0  broadcast 169.254.8.255
        inet6 fe80::690:81ff:fe00:1232  prefixlen 64  scopeid 0x20<link>
        ether 04:90:81:00:12:32  txqueuelen 1000  (Ethernet)
        RX packets 11  bytes 1038 (1.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 6  bytes 476 (476.0 B)
        TX errors 0  dropped 5 overruns 0  carrier 0  collisions 0
root@elba-7:/home/admin# ping 169.254.8.2
PING 169.254.8.2 (169.254.8.2) 56(84) bytes of data.
64 bytes from 169.254.8.2: icmp_seq=1 ttl=64 time=0.139 ms
64 bytes from 169.254.8.2: icmp_seq=2 ttl=64 time=0.130 ms
64 bytes from 169.254.8.2: icmp_seq=3 ttl=64 time=0.209 ms
```

### CODE REVIEW

File: patch/series [hunk 1]
```diff
@@ -253,6 +253,7 @@ armhf_secondary_boot_online.patch
 0028-EDAC-elba-Add-Elba-EDAC-support.patch
 0029-arm64-traps-Handle-SError-interrupt.patch
 0030-quirks-for-the-Pensando-qspi-controller.patch
+0031-platform-pensando-enable-ionic-x86.patch
```
Comment (saiarcot895) [inline patch/series:256]: You need to use `patch/kconfig-inclusions` to set/enable kconfig options.

Comment (shanshri) [reply patch/series]: Done

**Comment by KrisNey-MSFT** [comment]
Need to get AMD added to CLA for SONiC project (Sanjay)

**Comment by KrisNey-MSFT** [comment]
Process to gain approval for Individual CLA (for Shantanu to merge PR) (rather than Corporate CLA).

**Review by saiarcot895 — APPROVED** [review]

**Pull request merged**